src/main/java/org/lsc/jndi/JndiServices.java

```diff
@@ -745,28 +739,35 @@ public List<String> getDnList(final String base, final String filter,
 	private List<String> doGetDnList(final String base, final String filter,
 			final int scope) throws NamingException {
 		NamingEnumeration<SearchResult> ne = null;
-		List<String> iist = new ArrayList<String>();
+		List<String> list = new ArrayList<String>();
 		try {
 			SearchControls sc = new SearchControls();
 			sc.setDerefLinkFlag(false);
 			sc.setReturningAttributes(new String[]{"1.1"});
 			sc.setSearchScope(scope);
 			sc.setReturningObjFlag(true);
-			ne = ctx.search(base, filter, sc);
-			
-			String completedBaseDn = "";
-			if (base.length() > 0) {
-				completedBaseDn = "," + base;
-			}
-			while (ne.hasMoreElements()) {
-				iist.add(((SearchResult) ne.next()).getName() + completedBaseDn);
-			}
+			byte[] pagedResultsResponse = null;
+			do {
+				ne = ctx.search(base, filter, sc);
+				String completedBaseDn = "";
+				if (base.length() > 0) {
+					completedBaseDn = "," + base;
+				}
+				while (ne.hasMoreElements()) {
+					list.add(ne.next().getName() + completedBaseDn);
+				}
+				pagedResultsResponse = pagination();
+			} while (pagedResultsResponse != null);
+			contextRequestControls();
 		} catch (NamingException e) {
 			LOGGER.error(e.toString());
```

Comment:
in every catch structure, you should cal the clearsearchcontrols function to ensure the controls are removed (see my comment below for more details about this function)

Reply:
Using the try-catch-finally block to handle clearing the request controls.